src/liquid/components/ld-cookie-consent/ld-cookie-consent.tsx

```diff
@@ -336,21 +339,394 @@ export class LdCookieConsent {
+              mode="secondary"
+              onClick={() => {
+                this.showPreferences()
+              }}
+              part="disclaimer-button-preferences"
+              size="sm"
+            >
+              {config.buttonPreferences}
+            </ld-button>
+          )}
+          {config.mode !== 'notice-only' && config.rejectable && (
+            <ld-button
+              class="ld-cookie-consent__disclaimer-btn-accept-none"
+              mode="secondary"
+              onClick={() => {
+                this.hideDisclaimerAndPreferences()
+                this.saveConsent('accept-none')
+              }}
+              part="disclaimer-button-accept-none"
+              size="sm"
+            >
+              {config.buttonAcceptNone}
+            </ld-button>
+          )}
+          <ld-button
+            class={getClassNames([
+              config.mode === 'notice-only'
+                ? 'ld-cookie-consent__disclaimer-btn-acknowledge'
+                : 'ld-cookie-consent__disclaimer-btn-accept-all',
+            ])}
+            onClick={() => {
+              this.hideDisclaimerAndPreferences()
+              this.saveConsent('accept-all')
+            }}
+            part="disclaimer-button-accept-all"
+            size="sm"
+          >
+            {config.mode === 'notice-only'
+              ? config.buttonAcknowledge
+              : config.buttonAcceptCookies}
+          </ld-button>
+        </div>
+      </div>
+    )
+  }
+
+  private renderPreferences() {
+    const { config } = this
+    return (
+      <ld-modal
+        blurry-backdrop={config.backdropBlur}
+        class="ld-cookie-consent__preferences"
+        part="preferences"
+        ref={(ref) => (this.modalRef = ref)}
+      >
+        <div
+          slot="header"
+          class="ld-cookie-consent__preferences-header"
+          part="preferences-header"
+        >
+          <div
+            class="ld-cookie-consent__preferences-logo-container"
+            part="preferences-logo-container"
+          >
+            <slot name="preferences-logo">{this.getInitialM()}</slot>
+          </div>
+
+          <slot name="preferences-title">
+            <ld-typo
+              class="ld-cookie-consent__preferences-title"
+              part="preferences-title"
+              tag="p"
+              variant="h5"
+            >
+              Cookie Settings
+            </ld-typo>
+          </slot>
+        </div>
+
+        <div
+          class="ld-cookie-consent__preferences-description"
+          part="preferences-description"
+        >
+          <slot name="preferences-description" />
+        </div>
+
+        <ld-accordion
+          brand-color
+          class="ld-cookie-consent__preferences-accordion"
+          detached
+          part="preferences-accordion"
+          rounded
+          single
+        >
+          {config.categories.map(this.renderCategory.bind(this))}
```

Comment:
I don't think binding `this` is required here?
```suggestion
          {config.categories.map(this.renderCategory)}
```

Reply:
It is. If you remove the `bind`, the unit tests catch the error.